### PR TITLE
Add notes about sparsity.

### PR DESCRIPTION
**Pull request by jpountz**
cc @GlenRSmith

### CODE REVIEW

**Comment by jpountz** [comment]
@clintongormley any idea why I am getting `asciidoc: WARNING: general.asciidoc: line 58: section title out of sequence: expected level 2, got level 3 at lib/ES/Util.pm line 68.` when building docs with this PR?

**Comment by nik9000** [comment]
Docs look good to me. I'm not sure about the asciidoc warning though.

**Comment by clintongormley** [comment]
@jpountz all of your level 2 `===` headers are floated, so you can't use an unfloated level 3 `====` header

File: docs/reference/how-to/general.asciidoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+in the index have a timestamp field but some call it `timestamp` and others
+call it `creation_date`, it would help to rename it so that all documents have
+the same field name for the same data.
+
+==== Avoid types
+
+Types might sound like a good way to store multiple tenants in a single index.
+They are not: given that types store everything in a single index, having
+multiple types that have different fields in a single index will also cause
+problems due to sparsity as described above. If your types to not have very
```
Comment (tlrx) [inline docs/reference/how-to/general.asciidoc]: to not -> do not

**Comment by jpountz** [comment]
Thank you @nik9000 @clintongormley and @tlrx !

File: docs/reference/how-to/general.asciidoc [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+getting back the top documents that match a query. However, it is not as good
+for workloads that fall into the database domain, such as retrieving all
+documents that match a particular query. If you need to do this, make sure to
+use the <<search-request-scroll,Scroll>> API.
+
+[float]
+[[sparsity]]
+=== Avoid sparsity
+
+The data-structures behind Lucene, which elasticsearch relies on in order to
```
Comment (jasontedor) [inline docs/reference/how-to/general.asciidoc]: Nit: `elasticsearch` -> `Elasticsearch`

File: docs/reference/how-to/general.asciidoc [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+have access to the `norm` values of every document, this has the drawback that
+documents that do not have a value will also require one byte of storage.
+
+In practice, this means that if an index has `M` documents, norms will require
+`M` bytes of storage *per field*, even for fields that only appear in a small
+fraction of the documents of the index. Although slightly more complex with doc
+values due to the fact that doc values have multiple ways that they can be
+encoded depending on the type of field and on the actual data that the field
+stores, the problem is very similar. In case you wonder: `fielddata`, which was
+used in elasticsearch pre-2.0 before being replaced with doc values, also
```
Comment (jasontedor) [inline docs/reference/how-to/general.asciidoc]: Nit: `elasticsearch` -> `Elasticsearch`

File: docs/reference/how-to/general.asciidoc (hunk 2 above)
Comment (jasontedor) [inline docs/reference/how-to/general.asciidoc]: Oh, this is already merged. Sorry, I'll just push to master directly. 😇

File: docs/reference/how-to/general.asciidoc (hunk 3 above)
Comment (jasontedor) [inline docs/reference/how-to/general.asciidoc]: I'll push directly.

File: docs/reference/how-to/general.asciidoc (hunk 2 above)
Comment (jpountz) [inline docs/reference/how-to/general.asciidoc]: Thanks for having a look!

**Comment by GlenRSmith** [comment]
@jpountz Huge thanks for doing this!

**Pull request merged**